classifier_and_summarizer/summarizer/text_summarizer.py

```diff
@@ -0,0 +1,77 @@
+''' Module to perform text summarization'''
+
+from gensim.summarization.summarizer import summarize
+from summarizer import Summarizer
+
+from classifier_and_summarizer.summarizer.incorrect_input import \
+    IncorrectInputError
+
+
+class TextSummarizer:
+    """
+    A class to perform summarization on english text
+
+    Attributes:
+        - bertSummarizer : a bert-extractive-summarizer model
+        - textRankSummarizer : a summarizer based on text rank algorithm
+
+    Methods:
+        - summarizeText (text , accuracy , ratio):
+            Given the text and necessary accuracy and ratio
+            returns the summarized text
+        - textRankSummarization(text, ratio) :
+            applies textRankSummarizer to perform summarization and
+            returns the text
+        - bertSummarization(text , ratio) :
+            applies bert for summarization and returns the summarized text
+    """
+
+    def __init__(self):
+        '''
+        Initializes the text rank and bert summarizers
+        '''
+        self.bertSummarizer = Summarizer()
+        self.textRankSummarizer = summarize
+
+    def __textRankSummarization(self, text, ratio):
+        '''
+        Finds the summarization by applying Text Rank
+
+        Paramaters:
+            - text : text to be summarized
+            - ratio : a value in range [0,1] : what % of the text to retain
+
+        Returns : the summarized text
+        '''
+        return self.textRankSummarizer(text, ratio)
+
+    def __bertSummarization(self, text, ratio):
+        '''
+        Finds the summarization by applying Bert to find sentence embeddings
+        and then clusters the embeddings based on their similarity
+
+        Paramaters:
+            - text : text to be summarized
+            - ratio : a value in range [0,1] : what % of the text to retain
+
+        Returns :the summarized text
+        '''
+        return self.bertSummarizer(text, ratio)
+
+    def summarizeText(self, text, accuracy='low', ratio=0.4):
```

Comment:
Is there a reason why someone might opt for a "low" accuracy summarizer here instead of "high". Mainly, are there any advantages of using textRankSummarization over bertSummarization? I think the parameter here should be changed to reflect that more clearly. Or, we could simply use the names of the algorithms "textRank" or "bert" here instead of "high" and "low".

Reply:
The trade-off here is speed. Higher accuracy (bert) ends up being comparatively slower than textRank. Bert models are better able to capture meanings in the sentences .
Also i felt that allowing the parameter to be either "textRank" or "bert" would give away details as to its internal implementation - is this alright? Let me know what you think!

Reply:
I'm okay with not naming the parameter with the summarizer's name. But I think we should find a better way to differentiate between the 2 algorithms. As an end user why will choose the accuracy to be "low" unless I know that it helps with the speed?

Reply:
I agree. So perhaps shall we change it to 'latency' ? Once again i'm not sure how to indicate to the user that there is a trade-off between speed and accuracy.

Reply:
I guess we could add the parameter as `priority` and this can be either accuracy or speed This would indicate to the use that there is a tradeoff and one can have either good speed or good accuracy. Let me know what you think about this.

Reply:
This sounds good to me!